gneiss/balances.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import division
+import numpy as np
+from skbio.stats.composition import clr_inv
+from collections import OrderedDict
+
+
+def _balance_basis(tree_node):
+    """ Helper method for calculating balance basis
+    """
+
+    n_tips = sum([n.is_tip() for n in tree_node.traverse()])
```

Comment:
Here are the actual timings of `_balance_basis` as is

```python
10000 loops, best of 3: 43.5 µs per loop
```
And here are the timings if I put in the `n_tips` calculation within the second loop of the `_count_matrix` function.
```python
10000 loops, best of 3: 37.7 µs per loop
```

So not much difference at all.  I don't have a strong preference, so we can just do it the way you suggested.